diagnostics/app/model/diagnostics/csp/CSP.scala

```diff
@@ -1,24 +1,42 @@
 package model.diagnostics.csp
 
-import model.diagnostics.CSPDynamoDbReport
 import play.api.libs.functional.syntax._
 import play.api.libs.json._
+import common.Logging
+import net.logstash.logback.marker.Markers._
 
 case class CSPReport(
+  documentUri: String,
+  referrer: String,
   blockedUri: String,
-  documentUri: String)
+  violatedDirective: String,
+  effectiveDirective: String,
+  originalPolicy: String)
 
 object CSPReport {
   implicit val jsonReads: Reads[CSPReport] = (
-    (__ \ "blocked-uri").read[String] and
-      (__ \ "document-uri").read[String]
+    (__ \ "document-uri").read[String] and
+      (__ \ "referrer").read[String] and
+      (__ \ "blocked-uri").read[String] and
+      (__ \ "violated-directive").read[String] and
+      (__ \ "effective-directive").read[String] and
+      (__ \ "original-policy").read[String]
     )(CSPReport.apply _)
+
+  implicit val jsonWrites: Writes[CSPReport] = (
+    (__ \ "document-uri").write[String] and
+      (__ \ "referrer").write[String] and
+      (__ \ "blocked-uri").write[String] and
+      (__ \ "violated-directive").write[String] and
+      (__ \ "effective-directive").write[String] and
+      (__ \ "original-policy").write[String]
+    )(unlift(CSPReport.unapply))
```

Comment:
Yeah I know but here we are specifying every single field because our class parameters doesn't spell the same as the json fields